src/directives/tooltip.js

```diff
@@ -0,0 +1,329 @@
+Because aria-describedby and aria-label don't work as well with static elements,
+we usually place the additional text in its own element. That element will have
+the sr-only HTML class, which will hide the text off-screen while making it
+available to screen readers. To show the text of the .sr-only element as a
+tooltip on the first element (for example, on an icon), specify
+v-tooltip.sr-only. When the user hovers over or focuses the first element, the
+text of the .sr-only element will be shown in a tooltip.
+
+In rare cases, it may be that a particular tooltip does not work within any of
+the patterns above. For example, some text may be shown in a tooltip over a
+static element, then also made available on a nearby interactive element using
+an aria-describedby attribute. In cases like that, specify
+v-tooltip.no-aria="someValue". Doing so will pass the text to v-tooltip without
+connecting the text to an attribute or element available to screen readers.
+v-tooltip.no-aria should only be needed in rare cases and is intended as an
+escape hatch.
+
+If a value is passed directly to v-tooltip, using either
+v-tooltip.aria-describedby or v-tooltip.no-aria, it will be ignored if it is
+`null` or `undefined`: a tooltip will not be shown.
+
+In testing, use .should.textTooltip() to test v-tooltip.text. Use
+.should.tooltip() to test other uses of v-tooltip. In both cases, the assertion
+is asynchronous, so you must use `await`. For example:
+
+  await component.should.have.tooltip('Some text');
+*/
+
+import { computePosition, detectOverflow, flip, inline, offset } from '@floating-ui/dom';
+
+import { noop } from '../util/util';
+
+const types = ['text', 'aria-describedby', 'aria-label', 'sr-only', 'no-aria'];
+const placements = ['top', 'right', 'bottom', 'left'];
+const tooltips = new Map();
+let id = 1;
+
+// This Floating UI middleware is needed to horizontally center a v-tooltip.text
+// tooltip above or below its anchor element.
+const detectTextOverflow = {
+  name: 'detectTextOverflow',
+  fn: async (middlewareArgs) => {
+    const { placement } = middlewareArgs;
+    if (!(placement === 'top' || placement === 'bottom')) return {};
+    const overflow = await detectOverflow(middlewareArgs, {
+      elementContext: 'reference'
+    });
+    return {
+      // Accounting for both overflow.right and overflow.left with the thought
+      // that we will likely support RTL text in the future.
+      x: middlewareArgs.x -
+        (overflow.right > 0 ? overflow.right / 2 : 0) +
+        (overflow.left > 0 ? overflow.left / 2 : 0)
+    };
+  }
+};
+
+const show = (event) => {
+  const tooltip = tooltips.get(event.target);
+  if (!tooltip.shown) tooltip.show();
+};
+const hide = (event) => {
+  const tooltip = tooltips.get(event.target);
+  if (tooltip.shown) tooltip.hide();
+};
+
+class Tooltip {
+  static maybeCreate(anchor, { modifiers, value }) {
+    const type = types.find(name => modifiers[name] === true);
```

Comment:
how about just 
`const type = types.find(name => modifiers[name]);` or 
`const type = types.find(name => !!modifiers[name]);`

Reply:
Just personally, I try not to rely on truthiness and try to use explicit boolean expressions. Here, `modifiers[name]` may be `undefined`, so `modifiers[name]` isn't consistently boolean. I think issa has had a similar preference, so you'll see a lot of `=== true` and `=== false` in Frontend and maybe even more so in Backend. It's not a strong preference though! I certainly don't have any issues with others reaching for that pattern.